app/src/main/java/org/kiwix/kiwixmobile/zim_manager/fileselect_view/ZimFileSelectFragment.java

```diff
@@ -194,18 +195,14 @@ public boolean onActionItemClicked(ActionMode mode, MenuItem item) {
                       selectedViewPosition.get(i));
               String shareFilePath = data.file.getPath(); //Returns path to file in device storage
 
-              /**
-               * Using 'file:///' URIs directly is unsafe as it grants the intent receiving application
-               * the same file system permissions as the intent creating app.
-               *
-               * FileProvider instead offers 'content://' URIs for sharing files, which offer temporary
-               * access permissions to the file being shared (to the intent receiving application), which
-               * is fundamentally safer.
-               */
-
               File shareFile = new File(shareFilePath);
-              Uri shareContentUri = FileProvider.getUriForFile(getContext(),
-                  BuildConfig.APPLICATION_ID + ".fileprovider", shareFile);
+              Uri shareContentUri;
+              if (Build.VERSION.SDK_INT >= 24) {
+                shareContentUri = FileProvider.getUriForFile(Objects.requireNonNull(getContext()),
```

Comment:
As I have mentioned before use `zimManageActivity` wherever you're using context so change this line to `shareContentUri = FileProvider.getUriForFile(zimManageActivity,`